src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\Console\Helper\Table;
+
+/**
+ * Encode a user's password.
+ *
+ * @author Sarah Khalil <mkhalil.sarah@gmail.com>
+ */
+class UserPasswordEncoderCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('security:encode-password')
+            ->setDescription('Encode a password.')
+            ->setHelp(<<<EOF
+
+The <info>%command.name%</info> command allows to encode a password using encoders
+that are configured in the application configuration file, under the <comment>security.encoders</comment>.
+
+For instance, if you have the following configuration for your application:
+<comment>
+    security:
+        encoders:
+            Symfony\Component\Security\Core\User\User: plaintext
+            AppBundle\Model\User: bcrypt
+</comment>
+
+According to the response you will give to the question "<question>Provide your configured user class</question>" your
+password will be encoded the way it was configured.
+    - If you answer "<comment>Symfony\Component\Security\Core\User\User</comment>", the password provided will be encoded
```

Comment:
we should allow to pass the class name as an argument for a non-interactive usage.